sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -641,6 +641,12 @@ public final FileResult close() throws Exception {
       return result;
     }
 
+    @Override
+    public final void abort() throws Exception {
+      IOChannelUtils.deleteIfExists(filename);
+      close();
```

Comment:
For some file systems (like GCS), close is what actually creates the file.

you should close first, then delete with failure handling after close finishes.